code_submitter/auth.py

```diff
@@ -199,3 +204,43 @@ def __init__(self, data: List[NemesisUserInfo] = DEFAULT) -> None:
 
     async def load_user(self, username: str, password: str) -> NemesisUserInfo:
         return self.data[username]
+
+
+class FileBackend(BasicAuthBackend):
+    """
+    Authentication backend which stores credentials in a YAML file.
+
+    Credentials are stored in the format `TLA: password`.
+
+    Note: Passwords are stored in plaintext.
```

Comment:
I think it would be beneficial to call out that storing in plaintext is unusual and why we're happy with it?
Perhaps something like
```suggestion
    Credentials are stored in the format `TLA: password`.

    Note: Passwords are stored in plaintext.
    This is acceptable for some use-cases, for example where the data being
    uploaded is not sensitive or because someone having access to the
    credentials file almost certainly implies they have access to the upload
    storage anyway. However this may not be the case for all use-cases and you
    should evaluate the risks yourself before using this backend.
    You have been warned!
```